unitylibs/core/workflow/workflow-acrobat/action-binder.js

```diff
@@ -332,14 +410,20 @@ export default class ActionBinder {
     return true;
   }
 
-  async handleSingleFileUpload(file, eventName) {
-    const fileData = { type: file.type, size: file.size, count: 1 };
+  async handleSingleFileUpload(file, eventName) {  
+    const sanitizedFileName = this.sanitizeFileName(file.name); 
+
+    const newFile = new File([file], sanitizedFileName, { type: file.type, lastModified: file.lastModified });
+    const fileData = { type: newFile.type, size: newFile.size, count: 1 };
+
     this.dispatchAnalyticsEvent(eventName, fileData);
-    if (!await this.validateFiles([file])) return;
+
+    if (!await this.validateFiles([newFile])) return;
+    
     const { default: UploadHandler } = await import(`${getUnityLibs()}/core/workflow/${this.workflowCfg.name}/upload-handler.js`);
     this.uploadHandler = new UploadHandler(this, this.serviceHandler);
-    if (this.signedOut) await this.uploadHandler.singleFileGuestUpload(file, fileData);
-    else await this.uploadHandler.singleFileUserUpload(file, fileData);
+    if (this.signedOut) await this.uploadHandler.singleFileGuestUpload(newFile, fileData);
+    else await this.uploadHandler.singleFileUserUpload(newFile, fileData);
   }
 
   async handleMultiFileUpload(files, totalFileSize, eventName) {
```

Comment:
It looks like we are not sanitizing the file names for files in the multi-file upload use case. Can we please include this step here too?